### PR TITLE
Mouse Right Click Camera Panning Movement. 

### PR DESCRIPTION
**Pull request by TejasJD**
- Added Mouse Panning Support for camera movement (preliminary). Drag the Mouse holding the right button to move the editor camera.
- Updated the shortcut keys to switch between, Translation Rotation and Scale Gizmos.

### CODE REVIEW

**Review by alex-mulkerrins — APPROVED** [review]

**Review by prajakta-kulkarni-git — APPROVED** [review]
Looks good.

File: HorizonEditor/src/EditorLayer.cpp [hunk 1]
```diff
@@ -90,21 +87,22 @@ void EditorLayer::onUpdate(Hzn::TimeStep ts)
 
 	auto viewportSize = m_ViewportBounds[1] - m_ViewportBounds[0];
 
-	if(0 < mousePos.x && mousePos.x < viewportSize.x  && 0 < mousePos.y && mousePos.y < viewportSize.y)
-		HZN_INFO("{0}, {1}", mousePos.x, mousePos.y);
-
+	if (0 < mousePos.x && mousePos.x < viewportSize.x && 0 < mousePos.y && mousePos.y < viewportSize.y)
+	{
+		/*HZN_INFO("{0}, {1}", mousePos.x, mousePos.y);*/
```
Comment (LiamANeeson) [inline HorizonEditor/src/EditorLayer.cpp:92]: Why is the code within the comment unnecessary

File: HorizonEditor/src/EditorLayer.cpp [hunk 2]
```diff
@@ -711,17 +709,17 @@ bool EditorLayer::onKeyPressed(Hzn::KeyPressedEvent& e)
 		m_GizmoType = ImGuizmo::OPERATION::NONE;
 		break;
 	}
-	case Hzn::Key::E:
+	case Hzn::Key::W:
```
Comment (LiamANeeson) [inline HorizonEditor/src/EditorLayer.cpp:712]: Differentes in the new letters 

```
case Hzn::Key::E:
case Hzn::Key::R:
case Hzn::Key::T:
```

File: HorizonEditor/src/EditorLayer.cpp (hunk 1 above)
Comment (TejasJD) [reply HorizonEditor/src/EditorLayer.cpp:92]: This is for the next update.

File: HorizonEditor/src/EditorLayer.cpp (hunk 2 above)
Comment (TejasJD) [reply HorizonEditor/src/EditorLayer.cpp:712]: Because of changed camera controls.

**Review by LiamANeeson — APPROVED** [review]
Had a few general questions, overall tho great work!

**Pull request merged**